.github/workflows/push-to-prod.yml

```diff
@@ -5,87 +5,40 @@ on:
+  balena-cli: v12.52.1
 
 jobs:
-  outdoor-868:
+  build-prod:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        sbc: [raspi, rockpi]
+        frequency: [470, 868, 915]
+        variant: [indoor, outdoor]
     steps:
       - name: Checkout
         uses: actions/checkout@v2
+      - name: Change i2c bus for RockPi
+        if: matrix.sbc == 'rockpi'
+        run: |
+          sed -i -E "s/\/dev\/i2c-1:\/dev\/i2c-1/\/dev\/i2c-7:\/dev\/i2c-7/g" docker-compose.yml
+          ROCKPI="-rockpi"
+          echo "ROCKPI=$ROCKPI" >> $GITHUB_ENV
       - name: Balena Deploy
-        uses: nebraltd/balena-cli-action@v12.51.1
+        uses: nebraltd/balena-cli-action@${{ env.balena-cli }}
         if: success()
         with:
           balena_api_token: ${{secrets.BALENA_API_TOKEN}}
-          balena_command: "deploy nebraltd/helium-outdoor-868 --logs --debug --nocache --build"
-
-  indoor-868:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: Balena Deploy
-        uses: nebraltd/balena-cli-action@v12.51.1
-        if: success()
-        with:
-          balena_api_token: ${{secrets.BALENA_API_TOKEN}}
-          balena_command: "deploy nebraltd/helium-indoor-868 --logs --debug --nocache --build"
-
-  outdoor-915:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: Balena Deploy
-        uses: nebraltd/balena-cli-action@v12.51.1
-        if: success()
-        with:
-          balena_api_token: ${{secrets.BALENA_API_TOKEN}}
-          balena_command: "deploy nebraltd/helium-outdoor-915 --logs --debug --nocache --build"
-
-  indoor-915:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: Balena Deploy
-        uses: nebraltd/balena-cli-action@v12.51.1
-        if: success()
-        with:
-          balena_api_token: ${{secrets.BALENA_API_TOKEN}}
-          balena_command: "deploy nebraltd/helium-indoor-915 --logs --debug --nocache --build"
-
-  outdoor-470:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: Balena Deploy
-        uses: nebraltd/balena-cli-action@v12.51.1
-        if: success()
-        with:
-          balena_api_token: ${{secrets.BALENA_API_TOKEN}}
-          balena_command: "deploy nebraltd/helium-outdoor-470 --logs --debug --nocache --build"
-
-  indoor-470:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: Balena Deploy
-        uses: nebraltd/balena-cli-action@v12.51.1
-        if: success()
-        with:
-          balena_api_token: ${{secrets.BALENA_API_TOKEN}}
-          balena_command: "deploy nebraltd/helium-indoor-470 --logs --debug --nocache --build"
+          balena_command: "deploy nebraltd/helium-${{ matrix.variant }}-${{ matrix.frequency }}${{ env.ROCKPI }} --logs --debug --nocache --build"
```

Comment:
Why does RockPi need its own set of fleets? Is it just because of the `devices:  /dev/i2c-1:/dev/i2c-1` block, or something else too?

Reply:
@marvinmarnold it will also need different fleet variable for VARIANT and possibly some other custom stuff in the future.

Discussed with @vpetersson here - https://github.com/NebraLtd/helium-miner-software/issues/253#issuecomment-975639244

But perhaps you know of a better / simpler way of achieving this?

Reply:
@shawaj Will there also be a different variant for indoor vs outdoor rockpis? If not, the environment variable `RESIN_DEVICE_TYPE` can be used to detect  rockpi vs raspberry pi.

I'm going to look into the `/dev/i2c-X` issue now because I don't remember that being an issue for me.

The most elegant way to do this would be to write the variant to ECC during production (we've also talked about writing custom identify keys in there too). Alternatively, we could trigger [writing Balena device environment variables](https://www.balena.io/docs/reference/api/resources/device_environment_variable/) as part of the registration process.

Reply:
@marvinmarnold there will be an outdoor version, but whether we denote that as a separate variant I am not sure. I guess we probably should have it as a separate variant, because we will allow LTE modules with outdoor but not indoor.

Definitely agree it would be good to write arbitrary data such as variant and frequency to one of the data slots on the ECC during production, or potentially into the dedicated HAT EEPROM on the HAT for RockPi / RasPi. On the HAT, we also have the pin jumper setup that we can use if we want to. 

However, I don't know how we would get the data to determine what to program on ECC/EEPROM in this case - currently we determine it with a different image preloaded for a different app. Without that, there is no way to determine what it is from hardware or whatever. So whichever way we do it there will be a manual process somewhere along the line.

On the one hand, I don't like the idea of having lots of different fleets.

However on the other hand, it is quite a neat way of easily seeing / separating the different variants in balena with no further effort.

Not sure what you think.

Re the /dev/i2c-X issue - it wasn't an issue on the RockPi. It was an issue on the Raspberry Pi based units, when passing in /dev/i2c-7 which does not exist on raspberry pi (/dev/i2c-1 exists on rockpi so it still works)

Reply:
@shawaj 

> However, I don't know how we would get the data to determine what to program on ECC/EEPROM in this case 

Agreed that manufacturing will always need different images corresponding to different variants, but I'm not sure those images must also correspond to different fleets. I haven't been able to find a way to use inject different environment variables into the same base image. It may be possible to hijack [config.json](https://www.balena.io/docs/reference/OS/configuration/#sample-configjson) and insert a dummy SSH key or udev rule that really just indicates variant.

> On the one hand, I don't like the idea of having lots of different fleets. However on the other hand, it is quite a neat way of easily seeing / separating the different variants in balena with no further effort.

The most compelling reasons to keep different fleets IMHO are to reduce blast radius, limit access, and doing canary deployments. The lofi reporting (eg knowing how many outdoor 915mhz units we have) aspect is nice right now, but hopefully we move towards a more mature BI pipeline sooner than later. Also worth considering if Balena has scaling issues if we put everything in the same fleet.

That said, there aren't really huge reasons to prefer a single fleet. Searching for devices is the biggest benefit. And the fact that its the same software on all the devices just feels like everything belongs in one fleet.

> Re the /dev/i2c-X issue - it wasn't an issue on the RockPi. It was an issue on the Raspberry Pi based units

Gotcha, thanks.

Reply:
@marvinmarnold yeah I do agree overall on the single fleet, as well as the logo reporting not being super long term useful, and @vpetersson and I had discussed the very same thing you suggest about the config.json hijacking 👍. However I feel like it needs a bit more work to get to where we can do that.

Lastly, with regard to searching for devices we can do a lot of that through our dashboard as well so not the end of the world if it's less user friendly in balena.

What do you think re merging this PR anyway? Should I fix it up and merge or what are you thinking?

Reply:
Yea, lets merge this @shawaj. Multiple dockerfiles + fleets works good for now.

Reply:
👌 will fix up later this evening